lib/intMap.js

```diff
@@ -0,0 +1,51 @@
+var hasOwn = Object.prototype.hasOwnProperty;
+
+/**
+ * A very simple polyfill for Map with limited functionality
+ * that only works for integer keys.
+ */
+function IntMap() {
+    this.map = {};
```

Comment:
The perf gain is in not having to do the `hasOwnProperty` check but tbh like most micro-optimisations I doubt it's worth the hassle.